File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.swift
class Solution {
    func minCostClimbingStairs(_ cost: [Int]) -> Int {
        var dp = cost // Use the cost array as the DP table
        
        // Start the loop from the third step, since we can step from 0 or 1
        for i in 2..<cost.count {
            dp[i] = min(dp[i-1], dp[i-2]) + cost[i]
        }
        
        // The answer will be the minimum cost to reach either the last step or the step before it
        return min(dp[cost.count - 1], dp[cost.count - 2])
        
    }
}

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        
        n = len(cost)
        res=[0]*(n+1)
        if n==0:
            return 0
        if n==1:
            return cost[0]
        for i in range(2,n+1):
            res[i] = min(res[i-1]+cost[i-1], res[i-2]+cost[i-2])
        return res[-1]
            